Hivemind/Assets/Scripts/Gathering.cs

```diff
@@ -38,18 +57,46 @@ private void Awake()
 
     private ResourceNode findResource()
     {
-        ResourceNode resourceNode = GameWorld.FindNearestResource(transform.position, PreferredResource);
+        ResourceNode resourceNode = GameWorld.FindNearestKnownResource(transform.position, PreferredResource);
         if (PreferredResource != ResourceType.Unknown && resourceNode == null)
         {
-            resourceNode = GameWorld.FindNearestResource(transform.position, ResourceType.Unknown);
+            resourceNode = GameWorld.FindNearestKnownResource(transform.position, ResourceType.Unknown);
         }
         return resourceNode;
     }
 
+    private void TargetResource()
+    {
+        target = findResource();
+        if (target != null)
+        {
+            if (state == State.Idle)
+            {
+                agent.isStopped = false;
+            }
+            StopAllCoroutines();
+            scouting = false;
+            nextHarvest = target.DecreaseFutureResources(CarryAmount - carryingObjects.Count);
+            agent.SetDestination(target.GetPosition());
+            state = State.MovingToResource;
+        }
+        else if (state == State.Idle && IsScout)
+        {
+            agent.isStopped = false;
+            state = State.Scouting;
+        }
+        else if (Vector3.Distance(transform.position, storage.GetPosition()) > 2f)
+        {
+            state = State.MovingToStorage;
+        }
+    }
+
     private void carryResource(ResourceNode resource)
     {
         GameObject carryingObject = Instantiate(resource.baseObject.gameObject, transform.position, Quaternion.identity, transform);
         carryingObject.transform.localScale = new Vector3(carryingObject.transform.localScale.x * 3, carryingObject.transform.localScale.y * 3, carryingObject.transform.localScale.z * 3);
+        carryingObject.transform.position = new Vector3(transform.position.x, transform.position.y + 0.08f * (carryingObjects.Count + 1), transform.position.z);
+        carryingObject.transform.SetParent(gameObject.transform);
```

Comment:
shit dit was niet nodig geweest ik was een baka en had niet goed gekeken XD Instantiate doet dat paranten al en ik dacht dat je met positie de relatieve positie zet maar het is wel goed dat ie hier staat :P

Reply:
Er moet misschien alleen wel wat komen op die arbitrary 0.08f, als we de scale van objecten aanpassen of dit nou van de mieren zijn of van de resources dan klopt daar niet zo veel meer van, ik weet hier ook geen oplossing op, 1 regel hoger